ProcessLib/SmallDeformation/SmallDeformationFEM.h

```diff
@@ -493,6 +493,50 @@ class SmallDeformationLocalAssembler
         return *_ip_data[integration_point].material_state_variables;
     }
 
+    void computeSecondaryVariableConcrete(
+        double const /*t*/, std::vector<double> const& /*local_x*/) override
+    {
+        int const elem_id = _element.getID();
+        ParameterLib::SpatialPosition x_position;
+        x_position.setElementID(elem_id);
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        constexpr int symmetric_tensor_size =
+            MathLib::KelvinVector::KelvinVectorDimensions<
+                DisplacementDim>::value;
+
+        auto sigma_sum = MathLib::KelvinVector::tensorToKelvin<DisplacementDim>(
+            Eigen::Matrix<double, 3, 3>::Zero());
```

Comment:
Is there a relation between the DisplacementDim the fixed size 3 of the matrix?

Reply:
not in this case. We're always dealing with 3 normal stresses, from which we calculate the principal values.

Reply:
Thanks for answer.